### PR TITLE
Add support for GH actions

### PR DESCRIPTION
**Pull request by jaimergp**
GH Actions runs on Azure, so it will provide more build time (up to 6h) and Windows support.

# Checklist

- [x] Conda initializes properly
- [x] OpenEye license gets decrypted
- [x] Recipe builds
- [x] Test environment can be created
- [ ] Tests pass
- [ ] Deployment to Anaconda.org works

### CODE REVIEW

**Comment by jaimergp** [comment]
Who has the encryption key for the OE license? We need to add it to https://github.com/choderalab/perses/settings/secrets.

**Comment by jaimergp** [comment]
Tests run now, but there are some errors that I guess are unrelated to GHA.

**Comment by jchodera** [comment]
Thanks so much, @jaimergp !

@hannahbrucemacdonald and @dominicrufa : Can you review and merge?

**Comment by hannahbrucemacdonald** [comment]
I'm finding the GH test page really slow to respond/unresponsive which is making debugging very hard... does anyone else have this issue? In particular if I try search on the page

**Comment by jaimergp** [comment]
I guess these are particularly long logs because `perses` tests are a bit verbose (`Unknown bond types` etc). Depending on how much available RAM you have, the browser might have trouble keeping up (happened to me here amd in Azure and Travis). If might be worth trying inspecting the raw logs if that's the case:

image

**Comment by jchodera** [comment]
@dominicrufa @jaimergp : I think we've figured out how to fix failing tests in the other PRs. Some of the tests used paths relative to the current working directory, which is not a good practice. I've fixed these issues in #616 and #619, so I'd love to get this merged ASAP so we can start using the GH Actions reports!

**Comment by jaimergp** [comment]
I have converted the Travis skippers to GH Actions skippers. Other errors seem unrelated.

File: perses/tests/test_lambda_protocol.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from nose.tools import raises
 import os
 
-istravis = os.environ.get('TRAVIS', None) == 'true'
+running_on_github_actions = os.environ.get('GITHUB_ACTIONS', None) == 'true'
```
Comment (dominicrufa) [inline perses/tests/test_lambda_protocol.py]: is there a reason we aren't running this test?

**Comment by jchodera** [comment]
Is this ready to merge yet?

**Comment by jaimergp** [comment]
I think Hannah opened #621 to handle this!

**Comment by hannahbrucemacdonald** [comment]
Do we want tests passing before we merge? I can keep debugging tests, or we can merge as is

**Comment by jaimergp** [comment]
Are those errors related to GHA (misconfiguration, permissions, etc) or `perses`? I see lots of "file not found" errors, which could be due to `/tmp` not being writable in GHA (I honestly don't know, but we could workaround that setting `TMPDIR` to a writable location if needed).

**Comment by jchodera** [comment]
> Are those errors related to GHA (misconfiguration, permissions, etc) or perses? I see lots of "file not found" errors, which could be due to /tmp not being writable in GHA (I honestly don't know, but we could workaround that setting TMPDIR to a writable location if needed).

There appear to be a few different errors that were fixed in #616 and #619:
* Protocol keys:
```
======================================================================
ERROR: Ensure that we can instantiate and run a nonequilibrium relative free energy calculation for the cdk2 ligands in vacuum
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/miniconda/envs/test/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/runner/work/perses/perses/perses/tests/test_relative_setup.py", line 93, in test_run_cdk2_iterations_neq
    length_of_protocol = setup_options['n_steps_ncmc_protocol']
KeyError: 'n_steps_ncmc_protocol'
```
* `openforcefield` and some other dependencies need to be installed for testing, see [this](https://github.com/choderalab/perses/blob/master/.travis.yml#L61-L67):
```
======================================================================
ERROR: Ensure that we can instantiate and run a repex relative free energy calculation the cdk2 ligands in vacuum
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/miniconda/envs/test/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/runner/work/perses/perses/perses/tests/test_relative_setup.py", line 164, in test_run_cdk2_iterations_repex
    setup_dict = setup_relative_calculation.run_setup(setup_options)
  File "/home/runner/work/perses/perses/perses/app/setup_relative_calculation.py", line 336, in run_setup
    small_molecule_forcefield=setup_options['small_molecule_forcefield'], small_molecule_parameters_cache=setup_options['small_molecule_parameters_cache'])
  File "/home/runner/work/perses/perses/perses/app/relative_setup.py", line 216, in __init__
    from openforcefield.topology import Molecule
ModuleNotFoundError: No module named 'openforcefield'
-------------------- >> begin captured logging << --------------------
```
* This looks like an API error:
```
======================================================================
ERROR: Failure: TypeError (__init__() got an unexpected keyword argument 'scheme')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/miniconda/envs/test/lib/python3.6/site-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/share/miniconda/envs/test/lib/python3.6/site-packages/nose/loader.py", line 251, in generate
    for test in g():
  File "/home/runner/work/perses/perses/perses/tests/test_samplers.py", line 159, in test_hybrid_scheme
    testsystem.exen_samplers[environment] = ExpandedEnsembleSampler(testsystem.mcmc_samplers[environment], testsystem.topologies[environment], chemical_state_key, testsystem.proposal_engines[environment], geometry.FFAllAngleGeometryEngine(metadata={}), scheme='geometry-ncmc-geometry', options={'nsteps':1})
TypeError: __init__() got an unexpected keyword argument 'scheme'
-------------------- >> begin captured logging << --------------------
```
* This looks like a failure of `generateOEMolFromTopologyResidue`, which we're trying to expunge entirely from `perses` because going from elements and connectivity to a real `Molecule` is not robust:
```
======================================================================
ERROR: test the creation of unsampled endstates
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/share/miniconda/envs/test/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/runner/work/perses/perses/perses/tests/test_smc.py", line 242, in test_create_endstates
    phases = ['vacuum'])
  File "/home/runner/work/perses/perses/perses/app/relative_setup.py", line 117, in __init__
    all_old_mol = createSystemFromSMILES(self._ligand_smiles_old, title='MOL') # should be stereospecific
  File "/home/runner/work/perses/perses/perses/utils/openeye.py", line 198, in createSystemFromSMILES
    system, positions, topology = OEMol_to_omm_ff(molecule)
  File "/home/runner/work/perses/perses/perses/utils/openeye.py", line 132, in OEMol_to_omm_ff
    system = system_generator.build_system(topology)
  File "/home/runner/work/perses/perses/perses/rjmc/topology_proposal.py", line 2116, in build_system
    system = self._forcefield.createSystem(new_topology, **self._forcefield_kwargs)
  File "/usr/share/miniconda/envs/test/lib/python3.6/site-packages/simtk/openmm/app/forcefield.py", line 1135, in createSystem
    if generator(self, res):
  File "/usr/share/miniconda/envs/test/lib/python3.6/site-packages/openmoltools/forcefield_generators.py", line 514, in gaffTemplateGenerator
    molecule = generateOEMolFromTopologyResidue(residue)
  File "/usr/share/miniconda/envs/test/lib/python3.6/site-packages/openmoltools/forcefield_generators.py", line 148, in generateOEMolFromTopologyResidue
    infile = open(ac_output_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp4trmix4_/molecule-after-bond-perception.ac'
-------------------- >> begin captured logging << --------------------
```

**Comment by jchodera** [comment]
@hannahbrucemacdonald : Did we get this incorporated into another PR, or can we rebase here and get this merged? I can finish fixing the tests in #626

**Comment by hannahbrucemacdonald** [comment]
Lets merge

**Pull request merged**